apps/ui/src/lib/isLocalLink.ts

```diff
@@ -0,0 +1,10 @@
+export function isLocalLink(link: string): boolean {
+  try {
+    return (
+      link.startsWith("/") ||
+      window.location.hostname === new URL(link).hostname
+    )
+  } catch (error) {
+    return false
+  }
```

Comment:
to make this function compatible with both client and server side envs we need to avoid direct access to `window` object as it exists only in the browser. So I would slightly rework this:

```tsx
import { env } from "@/env.mjs"

...

  try {
    const baseUrl = env.NEXT_PUBLIC_APP_PUBLIC_URL
    const url = new URL(link, baseUrl)
    return url.hostname === new URL(baseUrl).hostname
  } catch (error) {
    // If URL constructor fails (invalid URL), treat it as external link
    return false
  }
```